apis/infrastructure/v1beta1/byocluster_webhook.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var byoclusterlog = logf.Log.WithName("byocluster-resource")
+
+func (r *ByoCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+//+kubebuilder:webhook:path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-byocluster,mutating=true,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=byoclusters,verbs=create;update,versions=v1beta1,name=mbyocluster.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &ByoCluster{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+// nolint: stylecheck
+func (r *ByoCluster) Default() {
+	byoclusterlog.Info("default", "name", r.Name)
+
+	if r.Spec.BundleLookupTag == "" {
+		r.Spec.BundleLookupTag = "v0.1.0_alpha.2"
```

Comment:
we'd better define an const for this and collect them together in a file and generate from current release

Reply:
+1 for reading from const

> collect them together in a file and generate from current release

This is still adding complexity IMO, we should be good with reading from an env variable while applying the cluster-template. Default value should be something like `latest` - But in our case, this default value may not be applicable yet

We can make this field **required** for now, it will error out if the field is not set. Unless we figure out a better way to publish bundles with `latest` tag to the repository, I suggest we leave it as is.

Reply:
I am aligned with @anusha94 suggestion as we haven't yet figured to tag appropriately the various bundles. Reading it for an env variable seems like a good option to me as of now.

Please note that currently we only have a single bundle for k8s 1.22.3 and ubuntu 20.04 but going forward we could have various bundles which might can increase the complexity of naming/tagging bundles.

Reply:
@anusha94 I accepted your comment. Please review it again. @dharmjit